draft-ietf-teep-protocol.md

```diff
@@ -337,7 +337,8 @@ token
   to distinguish the correct response from multiple requests.
   The token value MUST NOT be used for other purposes, such as a TAM to
   identify the devices and/or a device to identify TAMs or Trusted Components.
-  The TAM MUST expire the token value after receiving the first responce
+  The TAM SHOULD set the expiration term to each token and MUST ignore any messages with expired tokens.
```

Comment:
"set the expiration term" is hard to understand.  Do you mean "set an expiration timeout"?   Is there a recommended duration?  E.g., is an logically infinite expiration (say 20 years) ok or not ok?

Reply:
@dthaler I think it is ''set an expiration timeout" here. Do you have a recommendation of timeout duration? When teep-protocol is used over http, then probably about one day would be sufficient. When it is used over usb memory as described in suit usecases then one week or one month depending on TAM service provider until a service engineer visiting customer place and returning to the office. We could discuss further at IETF110 session.

Reply:
Isn't it "expiration time"? Timeout is different while "term" is too general.

As an expiration time needs to be associated with each token, is it a TAM policy for processing or should be part of the message along with token? Currently it implicitly assumes a TAM to store token creation time and knows when it expires when it receives a response from a TEEP Agent. This puts a burden of state management by TAM. A simpler approach is to carry that expiration time in the message to make the protocol stateless as much as possible. Well, matching a token between a request and response itself introduced a state requirement to TAM.

On expiration window, I think a default can be moderately small, say 24 hours. If a device doesn't respond with this long period, the TAM can query again. If a TAM is managing classes of devices that tend to be dormant for long, the TAM can decide a longer policy. This protocol doesn't need to dictate the value. On a security recommendation aspect, 20 years is surely too long for a 8 byte token value to sustain collision attack in my opinion.

Hope not expand the change scope too much at this stage of the spec.

Reply:
> Currently it implicitly assumes a TAM to store token creation time and knows when it expires when it receives a response from a TEEP Agent. This puts a burden of state management by TAM. A simpler approach is to carry that expiration time in the message to make the protocol stateless as much as possible. Well, matching a token between a request and response itself introduced a state requirement to TAM.

I would support any change to make the TAM as stateless as possible.
 
> Hope not expand the change scope too much at this stage of the spec.

Nope, please continue to discuss!

Reply:
To that end, we need to introduce a token expiration entry in the message, and specify its value timestamp format.

Reply:
I'm not sure that's needed.  The token is arbitrary format 8 to 64 bytes.  No reason the expiration can't be encoded inside the token itself.   If that can't be done in 64 bytes we can increase the max.  Point is that different TAM's may have different notions/formats for timestamp.  Since expiration is purely internal to a single TAM, there's no reason to standardize a format for it as far as I can tell.

Reply:
Hmm, it could work but isn't clean. In particular, current token definition says: "... is the same as in an EAT Nonce Claim (see {{I-D.ietf-rats-eat}} Section 3.3). The first usage of a token generated by a TAM MUST be randomly created.". It implies to be similar to a "nonce", and "is a random series of bits".